From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sat, 18 Nov 2017 20:00:58 +0100
Subject: [PATCH] {For LLVMLinux} Ensure -no-integrated-as is set when using
 as-instr

Overwise the clang assembler is tested for support of features, which
triggers error messages such as:

    lib/raid6/recov_avx512.c:387:2: error: "your version of binutils
    lacks AVX512 support" [-Werror,-W#warnings]
    #warning "your version of binutils lacks AVX512 support"
     ^

because:

    $ echo 'vpmovm2b %k1,%zmm5' | clang -c -x assembler -o /dev/null -
    <stdin>:1:1: error: instruction requires: AVX-512 BW ISA
    vpmovm2b %k1,%zmm5
    ^
---
 scripts/Kbuild.include | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/Kbuild.include b/scripts/Kbuild.include
index c75413d05a63..890897d69f85 100644
--- a/scripts/Kbuild.include
+++ b/scripts/Kbuild.include
@@ -106,7 +106,7 @@ as-option = $(call try-run,\
 # Usage: cflags-y += $(call as-instr,instr,option1,option2)
 
 as-instr = $(call try-run,\
-	printf "%b\n" "$(1)" | $(CC) $(KBUILD_AFLAGS) -c -x assembler -o "$$TMP" -,$(2),$(3))
+	printf "%b\n" "$(1)" | $(CC) $(KBUILD_AFLAGS) $(call cc-option, -no-integrated-as) -c -x assembler -o "$$TMP" -,$(2),$(3))
 
 # __cc-option
 # Usage: MY_CFLAGS += $(call __cc-option,$(CC),$(MY_CFLAGS),-march=winchip-c6,-march=i586)
-- 
